src/current/v23.1/changefeed-messages.md

```diff
@@ -23,35 +23,103 @@ This page describes the format and behavior of changefeed messages. You will fin
 
-When a changefeed targets a table with multiple column families, the family name is appended to the table name as part of the topic. See [Tables with columns families in changefeeds](changefeeds-on-tables-with-column-families.html#message-format) for guidance.
-
-For webhook sinks, the response format arrives as a batch of changefeed messages with a `payload` and `length`. Batching is done with a per-key guarantee, which means that messages with the same key are considered for the same batch. Note that batches are only collected for row updates and not [resolved timestamps](create-changefeed.html#resolved-option):
+For [webhook sinks](changefeed-sinks.html#webhook-sink), the response format arrives as a batch of changefeed messages with a `payload` and `length`. Batching is done with a per-key guarantee, which means that messages with the same key are considered for the same batch. Note that batches are only collected for row updates and not [resolved timestamps](create-changefeed.html#resolved-option):
 
 ~~~
 {"payload": [{"after" : {"a" : 1, "b" : "a"}, "key": [1], "topic": "foo"}, {"after": {"a": 1, "b": "b"}, "key": [1], "topic": "foo" }], "length":2}
 ~~~
 
-See [changefeed files](create-changefeed.html#files) for more detail on the file naming format for {{ site.data.products.enterprise }} changefeeds.
+Refer to [changefeed files](create-changefeed.html#files) for more detail on the file naming format for {{ site.data.products.enterprise }} changefeeds.
+
+## Message envelopes
+
+The _envelope_ defines the structure of a changefeed message. You can use the [`envelope`](create-changefeed.html#envelope) option to manipulate the changefeed envelope. The values that `envelope` accepts are compatible with different [changefeed sinks](changefeed-sinks.html), and the structure of the message will vary depending on the sink.
+
+{{site.data.alerts.callout_info}}
+Changefeeds created with [`EXPERIMENTAL CHANGEFEED FOR`](changefeed-for.html) or [`CREATE CHANGEFEED`](create-changefeed.html) with no sink specified (sinkless changefeeds) produce messages without the envelope metadata fields of changefeeds emitting to sinks.
+{{site.data.alerts.end}}
+
+The following sections provide examples of changefeed messages that are emitted when you specify each of the supported `envelope` options. Other [changefeed options](create-changefeed.html#options) can affect the message envelope and the what messages are emitted. Therefore, the examples are a guide for what you can expect when only the `envelope` option is specified.
+
+### `wrapped`
+
+`wrapped` is the default envelope structure for changefeed messages. This envelope contains an array of the primary key, a top-level field for the type of message, and the current state of the row (or `null` for [deleted rows](#delete-messages)). Kafka sinks do not contain an array of the primary key because the key is part of the message metadata.
```

Comment:
The array of the primary key is there when using a sink that doesn't have a message key as part of its protocol: cloud storage or webhook. Kafka and gcpubsub don't have it by default, but this is what the `key_in_value` option does when provided for those sinks.

Reply:
OK, I have added more detail on this and also included an example of `wrapped` + `key_in_value` for Kafka.